indico/modules/events/registration/util.py

```diff
@@ -677,6 +680,58 @@ def build_registration_api_data(registration):
     return registration_info
 
 
+def get_ticket_qr_code_data(person):
+    """Get the data which will be saved in a ticket QR code.
+
+    QR code format:
+
+    {
+        '': [qr_code_version, indico_url, b64(checkin_secret), b64(person_id)],
+        # extra keys may be added by plugins (e.g. site access)
+    }
+
+    This format tries to be as compact as possible so that the resulting QR codes
+    are small and easy to scan. However, we need to stick with a JSON dictionary in
+    order to be compatible with the site access plugin which inserts an extra key
+    with the ADaMS URL.
+
+    Note that `person_id` is only included if this is a ticket for an accompanying person.
+
+    The checkin secret and person id (if present) is base64-encoded to save space
+    (see https://stackoverflow.com/a/53136913/3911147).
+
+    If Indico is running on HTTPS, the scheme ('https://') is stripped from the
+    URL to save a few extra bytes.
+    """
+    registration = person['registration']
+    is_accompanying = person['is_accompanying']
+    person_id = person['id']
+    checkin_secret = person['registration'].ticket_uuid
+
+    qr_code_version = 2  # Increment this if the QR code format changes
+    url = _strip_scheme_if_https(config.BASE_URL)
+
+    data = {
+        '': [qr_code_version, url, _base64_encode_uuid(checkin_secret)]
```

Comment:
hmm are those 3 chars extra for `ind` (or 1 char for `i`) really a problem? dunno, empty key feels quite ugly,

Reply:
no real preference, we save a couple bytes for free (could make a difference for instances with longer URLs), but I agree that an empty key is not that pretty. I could go with `i`?

Reply:
yeah, let's go for `i` then